### DISCUSSION TITLE
⚱️ Archaeological Digging [PROPOSAL]

### DISCUSSION THREAD

**Discussion by adamhannigan** [sunflower-land/sunflower-land#4007] [Ideas]
## Overview

In the upcoming season, we will open an Archaeological site in the new "Desert area" above the beach. At the site, players will be able to use Sand Shovels + Drills to dig up items. The goal is to solve the puzzle each day, discover treasure and the hidden artefacts.

<img width="301" alt="Screenshot 2024-07-11 at 2 01 25 PM" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/5ebb9792-8840-485c-8056-588e6b9f5f1f">

**Disclaimer**

This is a proposal, not a feature specification. Requirements and mechanics will change up until launch of the feature. This is a draft design document.

## Puzzle

Instead of pure RNG, we will incorporate a puzzle element similar to Minesweeper & Battleship. Players will search the site for items. When you dig in a spot, you will find items which act as clues to discovering other items.

There will be 8 different items that can be found on site. Each item provides clues to the surrounding items.

<img width="209" alt="Screenshot 2024-07-11 at 2 45 33 PM" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/e7bbb9c0-3f82-469a-b68b-ad34afc88e7e">

For instance:

Sand = No item nearby
Crabs = Item in adjacent square
Seaweed = Always 3 in a vertical line, with Starfish at the bottom
Pipis = Always in a square
Sea Cucumber = 2 in horizontal line, with Starfish on left .... 

<img width="644" alt="Screenshot 2024-07-11 at 2 03 34 PM" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/e76f7ac2-7206-47bd-9a60-aaf3a88cdd44">

The pattern of the items will be placed around the Desert map - this will give you a hint (e.g. 3 pieces of seaweed with 1 starfish below will be located near the entrance)

**Desert Storm**

Each day at midnight UTC, a desert storm approaches and resets the archaeological site

## Goal

The goal each day is to discover treasures + the artefact in as little digs as possible.

- Treasure can be exchanged for Coins
- Sand will be used for Desert Island Expansions (12-20)
- Sand can be used to craft Sand Castles
- 2 Desert NPCs will appear, with delivery quests that require treasures (In exchange for Coins & Tickets)
- Players can swap the Artefact 
- Bert will occasionally request to see a treasure.

Treasure items are non-tradeable and non-withdrawable.

**Gambling Avoidance**

To comply with global gambling regulations, we ensure all of our features steer clear of any gambling connotations. In particular, the new digging design focuses on:

- Solving a puzzle: each player has their own puzzle that is generated each day they need to solve. Dig rewards are not random number generated.
- Treasure can only be sold for Coins (an in-game and non trade-able currency)

## Progression

So how will a player progress and attain mastery?

**Artefacts**

There will be 1 artefact available in the digging area each day. For our first season, the Artefact will be a "Scarab:. Each season a new artefact will be introduced, resetting the accumulation. Artefacts unused by the end of the season can be returned to the "Garbo".

Each day, the artefact will appear in a new and unique pattern. One day it may be surrounded by Pearls, the next day it may be surrounded by something else. We will provide visual clues near the site that indicates what pattern it will appear in.

A player will have the entire season to attain the artefacts and the prizes you can earn from them.

Reward 1 = 5 Artefacts
Reward 2 - 35 Artefacts
Reward 3 (Utility based) - 85 Artefacts
 
## MMO

This feature will take place in the MMO world:

1. Social pressure - if you see others digging, you will be more inclined. Especially if you see them find rewards.
2. Immersion - The puzzle and desert experience feels more immersive > moves away from slot machine mentality.

To avoid other players obstructing your view and grifting, we propose the following approach:

1. Other players who are actively 'on site' will be positioned **around the boundaries** (with a shovel icon above their head). When another player walks onto the site, they will 'poof' and appear on the edges. When they walk back off site (on their own game), they will poof and appear at the entrance again.
2. When a player discovers a treasure, it will pop up above their head. E.g. Coral icon
3. The current player will be the only one walking around the grid.

<img width="331" alt="Screenshot 2024-07-11 at 2 42 52 PM" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/c4c5907f-4c32-4b34-b115-973fce9fe47e">

## How to dig?

For this mechanic, we will implement touch (point to click) and path finding as the main controls.

Once a player enters the site, they can double click on a grid to shovel there.

<img width="737" alt="Screenshot 2024-07-11 at 2 42 20 PM" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/f3961e1e-f8d9-4b5d-942d-1f0d4f029294">

1. Player moves to that area
2. Digging animation begins
3. Reward shown (it stays there until next day)

_Why tap to move?_

We need to lock down a grid structure for digging and avoid choosing the wrong squares. Tapping feels like the most fluid and concrete method to do this.

**Sand Drill**

A sand drill can be used to dig up a 2x2 grid square.

**Dig Limits**

Once you have reached the dig limits a popup modal will show saying you are out of digs.
"5 more attempts = 1 Block Buck". This will serve as a new revenue stream for the project.

Since some items will be needed for seasonal achievements, we want to give the player as many chances as possible (They just need to pay extra Block Bucks + tools so they don't miss out)

**Artefact Shop**

The following items can be purchased at the Artefact shop. They are in-game items and non-tradeable.

Sand Shovel - 2 Wood, 1 Stone, 20 Coins
Sand Drill - 1 Oil, 1 Iron, 5 Wood, 40 Coins
Treasure Map - 5 Gold, 2 Wooden Compass
Bionic Drill (Wearable) - Extra 5 digs each day - 500 Sand, 10 Gold
Sand Castle 1 (Decoration) - 100 Sand
Sand Castle 2 (Decoration)

## Trade-able SFTs

Below are a range of collectibles & wearables that can help your digging abilities. These will be available throughout the season and trade-able.

Heart of Davy Jones: +20* Digs (Already Added ✅ )
Pharaoh Chicken: +2 Digs (Mutant Chicken Chance)
Ancient Shovel: Dig without shovel (Auction Supply 25) **OP Season Item**

## Extras

**Balancing**

We will adjust the prices of tools + treasures to coins based on beta testing to ensure it is an engaging feature. Overall, we aim for digging to burn more resources than it injects into the game.

**Bonus Idea - Meteorite**

Players will have a chance every day to find a Meteorite fragment. You can talk to X NPC, and give them 2 oil, to open the Meteorite. Inside players will find Stone, Iron, Gold or Gemstones.

**Bonus Idea -  Cave Area**

Unlock the mysterious Cave for bonus digging opportunities.

**Comment by TheDcol** [comment]
As someone who never liked the old TI aspect/gameplay, the puzzle aspect makes me really excited :)

**Comment by Dionis404** [comment]
That's a good idea, I love puzzles. 
I'm glad Heart of Davy Jones will be involved. 
What about the pirate's potion, can you think of something for it?

**Reply by adamhannigan** [reply]
We will reintroduce a Pirate NPC that will give players the Potion + Pirate Hat.

For the Pirate Potion, we will include the Chest. Current items include:

1 Sand Drill
2 Shovels
3 Shovels
5 Shovels

The purpose will be to give items that encourage more digging 🔨

**Comment by brian-man** [comment]
Artefact => Artifact?

**Reply by spencerdezartsmith** [reply]
Thats US spelling of `Artifact` 😝

**Reply by adamhannigan** [reply]
🇦🇺 > 🇺🇸 😜

**Comment by brian-man** [comment]
> Sand Shovel - 2 Wood, 1 Stone, 20 Coins
> Sand Drill - 1 Gold, 1 Iron, 40 Coins

Existing shovels and drills should be converted (up) to reflect the pricing change.

**Reply by brian-man** [reply]
@adamhannigan Thoughts on this one?

**Comment by brian-man** [comment]
> Sand will be used for Desert Island Upgrades

Will Sand be tradable? Is Sand available via any other source?

It sounds like using this feature is required for upgrading in Desert.  Not a big fan of "required minigames" for core game loops.

Also what are the desert upgrades?

**Reply by adamhannigan** [reply]
Sand will not be trade-able and only found through digging.

Desert Upgrades = Desert Expansions. I've updated the description.

It is important to see how new features build on top of our main gameplay loops (expansions + deliveries). This helps close the loop and give bonus utility for features.

I do hear your point around whether a minigame should be required or not. Currently you can expand with just the resources on your own land.

**Reply by brian-man** [reply]
@adamhannigan So are you saying that tile 26+ will be available in Desert? Or are you saying the expansion costs for the existing tiles up to 25 will be changed?

**Comment by Polysemouse** [comment]
I suggest showing a grid overlay once the player enters the digging arena

Will the puzzle be different for each player for the same day?

Will these treasures be removed from fishing?

Also I suggest writing unit tests extensively to make sure no weird patterns will be generated (eg. a line of 3 seaweed is not overwritten by another treasure on top of it / the same starfish does not correspond to a line of seaweeds and sea cucumbers)

**Reply by adamhannigan** [reply]
That is a good suggestion - a nice cue about the grid arranged.

Each player will get their own puzzle, generated at 12am UTC. This is to avoid players sharing the solution.

There will still be treasures available through fishing 👍 

Thanks for the suggestions on the unit tests - the algorithm isn't too complex so something we can get solid coverage over.

**Comment by brian-man** [comment]
Borrowing from Minesweeper, I'd like to be able to tag a grid location that I "know" already.

For example, if I find a pattern I may not want to spend digs uncovering (and collecting) the additional locations in the pattern.  But I want to mark them so that I can remember/plan.

Alternatively, I can just use a dry erase marker on my computer but that's kind of yuck.

**Reply by adamhannigan** [reply]
Interesting idea 👍 

This could be useful on mobile especially since you may not be able to view the entire screen

**Comment by brian-man** [comment]
> Ancient Shovel: Dig without shovel (Auction Supply 25) OP Season Item

Always a fan of "do X without Y" items.

Hate the idea of items with < 100 supply.
I'm guessing this means 10 BB, 5 SFL, 10 Tickets.  So auction rounds of 2 and 1 supply each.  Yuck.

**Reply by adamhannigan** [reply]
That is fair feedback. This is an extremely OP item, hence why we limited the supply. 

You make a great point about the small supply in each drop. It would be frustrating

**Comment by brian-man** [comment]
> Reward 3 (Utility based) - 85 Artefacts

The most recent season was not a full 3 months.  Assuming this pattern continues (or more break is added), the 85 should be tuned so it's not impossible.  Even with 90 days, demanding players succeed in collecting the artifact consistently for all but 5 days is a bit extreme IMO.

**Comment by drumdrumdrum3337** [comment]
Wasn’t the idea that the season will follow ancient Egypt theme? If that’s the case, then ideally digging should be following the theme.

For example, pipis and the like should be sunset (or left for fishing) and replaced with parts of old vases, pieces of mummy wraps, etc. The Cave in this case would become an inside of a pyramid.

Also, following the minesweeper analogy, should there be a “mine”? Some sort of an ancient curse that would make seaweed look like crabs for a day, or any other change that would make the puzzle more complicated to solve.

**Reply by adamhannigan** [reply]
Thanks @drumdrumdrum3337 

I agree that Egyptian items would be more suited for this area. We can brainstorm some items with the artists 👌 There may be a few items that can exist in both.

The **Mine** is also interesting 👌 Giving the players the opportunity to risk something is always fun gameplay. Adding this element of punishment would spice up the challenge

**Comment by SFLAustinHarris** [comment]
A few points:

-What specific upgrades does sand do for desert island?

-If sand castles and desert island upgrades are the only use case for sand, then shouldn't there be other map sections dedicated to other island upgrades the same way earning sand in the dig area is dedicated to desert island? (ex: exotic plants for petal paradise; volcanos for the upcoming lava island) I'm just trying to think how sand fits into the overall core game loop.

-Should oil be a requirement for sand drills and meteorites since they are not tradeable and can only be obtained through desert island? Similar to Brian's opinion, I don't think "luxury" items should be used for core game loops. I can't remember what was used for the sand drills in TI, but I'm almost sure it was just an expensive trade-able item like gold or iron.

**Reply by adamhannigan** [reply]
I've clarified the description. The intent was to use sand in the later Desert 'Expansions' (not upgrades).

I agree that core resources on an island should be used for upgrades. Fruit, Flowers + Honey would be ideal on Petal Paradise. We are going towards this approach 👍 

Good catch on the Oil, yes this will be part of Sand Drills. I missed it when typing the description 👌

**Comment by Jericho-rpc** [comment]
After changing the sand drill sand shovel recipe, is there any compensation for previous hoarders?

**Reply by adamhannigan** [reply]
The Sand Drill is now harder to craft. Those players who have already crafted, will keep their shovels 😄

**Comment by Thebyber** [comment]
One piece of information that has not been mentioned is the stock of sand shovels in the shop. I think it should be a stock that covers at least one day's excavation, having the different items that give you more daily excavations. In the old TI it was tedious to have to spend 2 BB to do 50 digs... since there were only 25-30 shovels of sand in the shop.

**Reply by adamhannigan** [reply]
Agreed 👍
